inc/classes/optimization/class-abstract-optimization.php

```diff
@@ -38,32 +38,58 @@ protected function find( $pattern, $html ) {
 	 * @return bool True if external, false otherwise
 	 */
 	protected function is_external_file( $url ) {
-		$file       = get_rocket_parse_url( $url );
-		$wp_content = get_rocket_parse_url( WP_CONTENT_URL );
-		// This filter is documented in inc/classes/admin/settings/class-settings.php.
-		$hosts   = apply_filters( 'rocket_cdn_hosts', [], $this->get_zones() );
-		$hosts[] = $wp_content['host'];
-		$langs   = get_rocket_i18n_uri();
+		$file = get_rocket_parse_url( $url );
+
+		if ( empty( $file['path'] ) ) {
+			return true;
+		}
+
+		/**
+		 * Filters the allowed hosts for optimization
+		 *
+		 * @since  3.4
+		 * @author Remy Perona
+		 *
+		 * @param array $hosts Allowed hosts.
+		 * @param array $zones Zones to check available hosts.
+		 */
+		$hosts      = apply_filters( 'rocket_cdn_hosts', [], $this->get_zones() );
+		$wp_content = get_rocket_parse_url( content_url() );
```

Comment:
What if `get_rocket_parse_url()` doesn't return an array that's has the right keys/structure?

Reply:
Done in https://github.com/wp-media/wp-rocket/commit/164c6c444a5d38449782299c9a91a7541313db94